docs/architecture/adr-020-protobuf-transaction-encoding.md

```diff
@@ -137,72 +187,106 @@ message SignDocRaw {
-the regular `SignDoc` type should be used.
+#### `SIGN_MODE_DIRECT_AUX`
 
-3. Broadcast `TxRaw`
+`SIGN_MODE_DIRECT_AUX` is used to support scenarios where multiple signatures
+are being gathered into a single transaction but the message composer does not
+yet know which signatures will be included in the final transaction. For instance,
+I may have a 3/5 multisig wallet and want to send a `TxBody` to all 5
+signers to see who signs first. As soon as I have 3 signatures then I will go
+ahead and build the full transaction.
 
-In order to make sure that the signed body bytes exactly match the encoded body
-bytes, clients should encode and broadcast `TxRaw` with the same body bytes used
-in `SignDocRaw`.
+With `SIGN_MODE_DIRECT`, each signer needs
+to sign the full `AuthInfo` which includes the full list of all signers and
+their signing modes, making the above scenario very hard.
+
+`SIGN_MODE_DIRECT_AUX` allows "auxiliary" signers to create their signature
+using only `TxBody` and their own `PublicKey`. This allows the full list of
+signers in `AuthInfo` to be delayed until signatures have been collected.
+
+An "auxiliary" signer can be any signer who is not paying a fee. For signers
+paying fees, the full `AuthInfo` is actually needed to calculate gas and fees
+because that is dependent on how many signers and which key types and signing
+modes they are using. Auxiliary signers, however, do not need to worry about
+fees or gas and thus can just sign `TxBody`.
+
+To generate a signature in `SIGN_MODE_DIRECT_AUX` follow these steps:
+
+1. Encode `SignDocAux` (with the same requirement that fields must be serialized
+in order):
 
 ```proto
 // types/types.proto
-message TxRaw {
-    bytes body_bytes = 1;
-    repeated bytes signatures = 2;
+message SignDocAux {
+    TxBody body = 1;
+    // PublicKey is included in SignDocAux :
+    // 1. as a special case for multisig public keys to be described later
+    // in this document
+    // 2. to guard against scenario where configuration information is encoded
+    // in public keys (it has been proposed) such that two keys can generate
+    // the same signature but have different security properties
+    //
+    // By including it here, the composer of AuthInfo cannot reference the
+    // a public key variant the signer did not intend to use
+    PublicKey public_key = 2;
+    string chain_id = 3;
+    uint64 account_number = 4;
+    // account_sequence starts at 1 rather than 0 to avoid the case where
+    // the default 0 value must be omitted in protobuf serialization
+    uint64 account_sequence = 5;
 }
 ```
 
-Signature verifiers should verify signatures by decoding `TxRaw` and then
-encoding `SignDocRaw` with the raw body bytes.
+2. Sign the encoded `SignDocAux` bytes
+3. Send their signature and `SignerInfo` to the tx composer(s) who will then
+sign and broadcast the final transaction (with `SIGN_MODE_DIRECT` and fees and
+gas added) once enough signatures have been collected
```

Comment:
Why would the tx composer be a signer? It can be a moderator/secretary/notary/bot that is just responsible for composing a transaction, collection the signatures, putting everything together and broadcasting it.

Reply:
So maybe this can be the primary signer who is the fee payer. The composer can be anyone. And anyone who had collected enough auxiliary signatures could sign the message with a fee.